### ISSUE TITLE
metric.MeasurementValue needs a ValueKind tag OR metric.Measurement cannot export fields

### ISSUE THREAD

**Issue by jmacd** [open-telemetry/opentelemetry-go#196]
Presently, the Measurement struct allows the user to supply batch measurements directly:
```
type Measurement struct {
	Descriptor *Descriptor
	Value      MeasurementValue
}
```

This allows a float measurement value to enter an integer metric instrument. We should either provide constructors for `Measurement` or include the information in MeasurementValue.

The OpenCensus library did the former, more-or-less. You could imagine a method on the instruments to construct measurements.  In OC this method was `M()`.  Then, `meter.RecordBatch(ctx, labels, []metric.Measurement{ instrument1.M(1), instrument2.M(2), instrument3.M(3) })`.

Alternatively, `MeasurementValue` can become

```
struct {
  uint64 raw
  kind ValueKind
}
```
The former is more type-safe. The latter makes it possible to truncate a floating point to an integer w/o realizing it.

**Comment by jmacd** [comment]
I'll add that I prefer the use of constructors for Measurement, but I do not have a good method name in mind.

**Comment by jmacd** [comment]
@krnowak This would make it safe to always assume the value passed to a metric handler matches its descriptor. Otherwise we still need coercion, and the MeasurementValue will need to know its own type.

**Comment by krnowak** [comment]
The assumption was that the ValueKind was stored in the descriptor and the measurement value should be accompanied with a descriptor which certainly is true for the `Measurement` struct. I need to reread the metrics specs as I am maybe missing some context now.

**Comment by jmacd** [comment]
The case I am concerned about would be a integer instrument for which a floating-point measurement is passed using `RecordBatch`.  If the user can construct `Measurement{}` values directly, the could pass a floating-point `MeasurementValue` and it would convert to the `uint64(math.Float64bits())` representation, I think.

**Comment by krnowak** [comment]
Normally I would expect the user to do something like:

```go
counter := meter.NewIntCounter("foo")
gauge := meter.NewFloatGauge("bar")
meter.RecordBatch(meter.DefineLabels(…), counter.Measurement(42), gauge.Measurement(13.26))
```

Maybe `Measurement` should only be created by instruments?

**Comment by jmacd** [comment]
> Maybe `Measurement` should only be created by instruments?

Yes, that will do it.

**Comment by krnowak** [comment]
Can this be closed? It was fixed in #228.
